### DISCUSSION TITLE
Could not create clip from recordings for {camera}

### DISCUSSION THREAD

**Discussion by simondrake** [blakeblackshear/frigate#4712] [Ask A Question]
**Note**
=======
This started as a question but along the way I figured out what the problem was (🦆) so, instead of just deleting it and moving on, I'm posting it incase others find it in their search for the same issue.  

**The Problem**
=============
I'm trying to download a clip from two days ago (2022-12-12) from between 12:10:40 (1670803840) and 12:10:50 (1670803850).

When hitting the endpoint (`http://<host>:5000/api/back_garden/start/1670803840/end/1670803850/clip.mp4`) I get a `Could not create clip from recordings for back_garden.` error in the browser with the following in the frigate logs:

```
[2022-12-14 10:25:59] frigate.http                   ERROR   : ffmpeg version n5.1-2-g915ef932a3-20220731 Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 12.1.0 (crosstool-NG 1.25.0.55_3defb7b)
  configuration: --prefix=/ffbuild/prefix --pkg-config-flags=--static --pkg-config=pkg-config --cross-prefix=x86_64-ffbuild-linux-gnu- --arch=x86_64 --target-os=linux --enable-gpl --enable-version3 --disable-debug --enable-iconv --enable-libxml2 --enable-zlib --enable-libfreetype --enable-libfribidi --enable-gmp --enable-lzma --enable-fontconfig --enable-libvorbis --enable-opencl --enable-libpulse --enable-libvmaf --enable-libxcb --enable-xlib --enable-amf --enable-libaom --enable-libaribb24 --enable-avisynth --enable-libdav1d --enable-libdavs2 --disable-libfdk-aac --enable-ffnvcodec --enable-cuda-llvm --enable-frei0r --enable-libgme --enable-libass --enable-libbluray --enable-libjxl --enable-libmp3lame --enable-libopus --enable-mbedtls --enable-librist --enable-libtheora --enable-libvpx --enable-libwebp --enable-lv2 --enable-libmfx --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopenmpt --enable-librav1e --enable-librubberband --disable-schannel --enable-sdl2 --enable-libsoxr --enable-libsrt --enable-libsvtav1 --enable-libtwolame --enable-libuavs3d --enable-libdrm --enable-vaapi --enable-libvidstab --enable-vulkan --enable-libshaderc --enable-libplacebo --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxvid --enable-libzimg --enable-libzvbi --extra-cflags=-DLIBTWOLAME_STATIC --extra-cxxflags= --extra-ldflags=-pthread --extra-ldexeflags=-pie --extra-libs='-ldl -lgomp' --extra-version=20220731
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
  libpostproc    56.  6.100 / 56.  6.100
[concat @ 0x560ef85d2a80] No files to concat
/dev/stdin: Invalid data found when processing input
```

**The Solution**
=============

The `No files to concat` error was pretty self explanatory, which is why I kept looking, but it was annoying as I was able to view the recording through the Frigate UI so the file was definitely there. 

My recordings for this camera are set to enabled with a retention of two days and, at the time of writing, two days had not passed to the hour so there was no reason the files wouldn't exist.

```yaml
record:
  enabled: True
  retain:
    days: 2
  events:
    retain:
      default: 10
```

TL;DR When I was converting from date/time to epoch, I had stupidly forgot to ensure I was using PM instead of AM, and two days HAVE passed since the AM time.

TL;DR TL;DR - [PICNIC](https://en.wiktionary.org/wiki/PICNIC)

**Comment by simondrake** [comment]
Side note - is there any appetite to have a Frigate CLI to handle this sort of operation (downloading clips)? It's something I can work on if so...

@NickM-27 @blakeblackshear

**Reply by blakeblackshear** [reply]
I don't think it's necessary. You can pass the existing m3u8 endpoint as an input to ffmpeg and create video files easily.

**Reply by blakeblackshear** [reply]
I definitely recommend using the m3u8 endpoint to ffmpeg instead of the clip.mp4 endpoint. It avoids the need for frigate to write a temp file to cache.

**Reply by simondrake** [reply]
Oh nice, I'll look into that then! 

Then I've just got to figure out why there's some sort of offset when using that endpoint. 

Thanks!

**Comment by MrCaspan** [comment]
I am having the exact same issue BUT I have 14 days retention and I am only trying to download a clip from 2 days ago. I get the same error in the log

```
2024-01-03 18:36:34.419605247  [2024-01-03 18:36:34] frigate.http                   ERROR   : ffmpeg version n5.1-2-g915ef932a3-20220731 Copyright (c) 2000-2022 the FFmpeg developers
2024-01-03 18:36:34.419606829    built with gcc 12.1.0 (crosstool-NG 1.25.0.55_3defb7b)
2024-01-03 18:36:34.419614217    configuration: --prefix=/ffbuild/prefix --pkg-config-flags=--static --pkg-config=pkg-config --cross-prefix=x86_64-ffbuild-linux-gnu- --arch=x86_64 --target-os=linux --enable-gpl --enable-version3 --disable-debug --enable-iconv --enable-libxml2 --enable-zlib --enable-libfreetype --enable-libfribidi --enable-gmp --enable-lzma --enable-fontconfig --enable-libvorbis --enable-opencl --enable-libpulse --enable-libvmaf --enable-libxcb --enable-xlib --enable-amf --enable-libaom --enable-libaribb24 --enable-avisynth --enable-libdav1d --enable-libdavs2 --disable-libfdk-aac --enable-ffnvcodec --enable-cuda-llvm --enable-frei0r --enable-libgme --enable-libass --enable-libbluray --enable-libjxl --enable-libmp3lame --enable-libopus --enable-mbedtls --enable-librist --enable-libtheora --enable-libvpx --enable-libwebp --enable-lv2 --enable-libmfx --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopenmpt --enable-librav1e --enable-librubberband --disable-schannel --enable-sdl2 --enable-libsoxr --enable-libsrt --enable-libsvtav1 --enable-libtwolame --enable-libuavs3d --enable-libdrm --enable-vaapi --enable-libvidstab --enable-vulkan --enable-libshaderc --enable-libplacebo --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxvid --enable-libzimg --enable-libzvbi --extra-cflags=-DLIBTWOLAME_STATIC --extra-cxxflags= --extra-ldflags=-pthread --extra-ldexeflags=-pie --extra-libs='-ldl -lgomp' --extra-version=20220731
2024-01-03 18:36:34.419615825    libavutil      57. 28.100 / 57. 28.100
2024-01-03 18:36:34.419617001    libavcodec     59. 37.100 / 59. 37.100
2024-01-03 18:36:34.419618405    libavformat    59. 27.100 / 59. 27.100
2024-01-03 18:36:34.419619715    libavdevice    59.  7.100 / 59.  7.100
2024-01-03 18:36:34.419621055    libavfilter     8. 44.100 /  8. 44.100
2024-01-03 18:36:34.419622709    libswscale      6.  7.100 /  6.  7.100
2024-01-03 18:36:34.419623774    libswresample   4.  7.100 /  4.  7.100
2024-01-03 18:36:34.419624816    libpostproc    56.  6.100 / 56.  6.100
2024-01-03 18:36:34.419625896  [concat @ 0x5629c335ba80] No files to concat
2024-01-03 18:36:34.419627035  /dev/stdin: Invalid data found when processing input
2024-01-03 18:36:34.419627971  
```
Any clue why i cannot export this 1 hour clip? i even tried a shorter clip with the same error

This is the URI I am using:

`http://192.168.86.232:30058/api/HouseFront/start/1704128400000/end/1704130000000/clip.mp4`

**Reply by didi767** [reply]
Hi @MrCaspan  did you manage to resolve this issue? facing the same one right now.

**Reply by MrCaspan** [reply]
I don't remember to be honest. I don't know if I gave up or it was fixed. Sorry, not the answer you want to hear. I'll have to test
